src/Symfony/Component/Validator/Constraints/NoBannedWordsValidator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class NoBannedWordsValidator extends ConstraintValidator
+{
+    private const LEET_MAP = [
+        'a' => '(a|4|/\\|@|\^|aye|∂|/\-\\|\|\-\\|q)',
+        'b' => '(b|8|6|13|\|3|ß|P\>|\|\:|\!3|\(3|/3|\)3)',
+        'c' => '(c|\(|¢|\<|\[|©)',
+        'd' => '(d|\[\)|\|o|\)|I\>|\|\>|\?|T\)|\|\)|0|\</)',
+        'e' => '(e|3|&|€|£|є|ë|\[\-|\|\=\-)',
+        'f' => '(f|\|\=|ƒ|\|\#|ph|/\=)',
+        'g' => '(g|6|&|\(_\+|9|C\-|gee|\(γ,)',
+        'h' => '(h|\#|/\-/|\[\-\]|\]\-\[|\)\-\(|\(\-\)|\:\-\:|\|~\||\|\-\||\]~\[|\}\{|\?|\}|\-\{|hèch)',
+        'i' => '(i|1|\!|\||\]\[|eye|3y3|\]|\:)',
+        'j' => '(j|_\||_/|¿|\</|\(/|ʝ| ;)',
+        'k' => '(k|X|\|\<|\|\{|ɮ|\<|\|\\“)',
+        'l' => '(l|1|£|1_|ℓ|\||\|_|\]\[_,)',
+        'm' => '(m|\|v\||\[V\]|\{V\}|\|\\/\||/\\/\\|\(u\)|\(V\)|\(\\/\)|/\|\\|\^\^|/\|/\||//\.|\.\\|/\^\^\\|///|\|\^\^\|)',
+        'n' => '(n|\^/|\|V|\|\\\||/\\/|\[\\\]|\<\\\>|\{\\\}|\]\\\[|//|\^|\[\]|/V|₪)',
+        'o' => '(o|0|\(\)|oh|\[\]|¤|°|\(\[\]\))',
+        'p' => '(p|\|\*|\|o|\|º|\|\^\(o\)|\|\>|\|"|9|\[\]D|\|̊|\|7|\?|/\*|¶|\|D)',
+        'q' => '(q|\(_,\)|\(\)_|0_|°\||\<\||0\.)',
```

Comment:
the brackets should just be removed actually

Reply:
Unfortunately, the behaviour is not the same with [] instead of ().
As an example, "$yMph0NY" is not detected because of "ph". Same if I change "symfony" into "symfohny". It looks like multicharacters are not understood when grouping with []

Reply:
I asked to use non-capturing groups, not character classes